src/main/java/frc/robot/subsystems/climb/Climb.java

```diff
@@ -0,0 +1,62 @@
+package frc.robot.subsystems.climb;
+
+import static frc.robot.subsystems.climb.ClimbConstants.MECHANISM_HEIGHT;
+import static frc.robot.subsystems.climb.ClimbConstants.MECHANISM_WIDTH;
+
+import edu.wpi.first.wpilibj.Timer;
+import edu.wpi.first.wpilibj.smartdashboard.Mechanism2d;
+import edu.wpi.first.wpilibj.smartdashboard.MechanismLigament2d;
+import edu.wpi.first.wpilibj.smartdashboard.MechanismRoot2d;
+import edu.wpi.first.wpilibj2.command.Command;
+import edu.wpi.first.wpilibj2.command.Commands;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import java.util.function.DoubleSupplier;
+import org.littletonrobotics.junction.AutoLogOutput;
+import org.littletonrobotics.junction.Logger;
+
+public class Climb extends SubsystemBase {
+
+    private static Climb INSTANCE;
+
+    private final ClimbInputsAutoLogged inputs = ClimbIO.inputs;
+    private final ClimbIO io;
+    private final Timer timer = new Timer();
+
+    @AutoLogOutput
+    private final Mechanism2d mechanism2d = new Mechanism2d(MECHANISM_WIDTH, MECHANISM_HEIGHT);
+
+    private final MechanismRoot2d root = mechanism2d.getRoot("Elevator", 0, 0);
+    private final MechanismLigament2d elevator =
+            root.append(new MechanismLigament2d("Elevator", 0, 90));
+
+    private Climb(ClimbIO io) {
+        this.io = io;
+
+        timer.start();
+        timer.reset();
+    }
+
+    public static Climb getInstance() {
+        return INSTANCE;
+    }
+
+    public static void initialize(ClimbIO io) {
+        INSTANCE = new Climb(io);
+    }
+
+    public Command manualClimb(DoubleSupplier power) {
+        return Commands.parallel(run(() -> io.setPower(power.getAsDouble())));
+    }
+
+    public Command manualReset() {
+        return Commands.runOnce(io::manualReset);
+    }
+
+    @Override
+    public void periodic() {
+        io.updateInputs(inputs);
+        if (timer.advanceIfElapsed(0.1)) {
+            Logger.processInputs(this.getClass().getSimpleName(), inputs);
+        }
```

Comment:
I really wouldn't do this. This needs to be run every loop for replay to work correctly. If anything, there is no reason to call `updateInputs` more frequently than `processInputs` -- all the latter does is write the inputs to the log (or read from it during replay). So I would remove this timer or if you're really sure it's necessary, put `updateInputs` inside the timer check too.

Reply:
I don't agree, you need to update the inputs every loop for them to be accurate but you don't need to process them at the same frequency.

Reply:
After thinking about it again you're correct about the first part (indeed there are other things that need updated inputs other than `processInputs()`), but there is still no reason to not call `processInputs()` in every loop. We've seen that what takes more time is calls to motor controllers and sensors, writing to the log file is a cheap operation.